src/hooks/useDatabase.ts

```diff
@@ -0,0 +1,81 @@
+import { Collection, IApplication, IModelBase, Status } from "~/@types/common";
+
+// taken from https://stackoverflow.com/a/63029283
+type DropFirst<T extends unknown[]> = T extends [any, ...infer U] ? U : never;
+
+const FREE_ID_KEY = 'free-id';
+
+const useDatabase = () => {
+  const get = <T extends IModelBase>(
+    collection: Collection,
+    filterFunction?: (entry: T) => boolean,
+  ): T[] => {
+    const entries: T[] = JSON.parse(localStorage.getItem(collection) || '[]');
+
+    return filterFunction ? entries.filter(filterFunction) : entries;
+  };
```

Comment:
Those `get`, `getById`, `insert`, `removeById`, `updateById` functions are generic - can be used with every Data Model.